src/components/widgets/Form.js

```diff
@@ -0,0 +1,315 @@
+import React, { Component } from 'react'
+import { RadioGroup, Radio } from 'react-radio-group'
+import Button from '../Button.js'
+import "../../styles/form.css"
+
+export default class Form extends Component {
+  constructor( props ) {
+    super( props )
+    this.state = {}
+    this.setFieldState = this.setFieldState.bind( this )
+  }
+
+  componentWillMount() {
+    this.setupState()
+  }
+
+  setupState() {
+    const { sections } = this.props
+    sections.forEach( section => {
+      section.fields.forEach ( field => {
+        const { name, defaultValue } = field
+        this.setState({[ name ]: defaultValue })
+        if ( field.type === "nested-dropdown" ) {
+          const nestedDropdownValue = field.options.filter( option => {
+            return option.value === defaultValue
+          })[0].field
+          this.setState({
+            [ nestedDropdownValue.name ]: nestedDropdownValue.defaultValue
+          })
+        }
+      })
+    })
+  }
+
+  setFieldState( state, stateValue ) {
+    this.setState({
+      [ state ]: stateValue
+    })
+  }
+
+  submitWidget( state ) {
```

Comment:
The name `submit` is fine here (we wouldn't be submitting anything else from this component :P)